csrc/id_model/indexing.h

```diff
@@ -59,7 +61,9 @@ class TensorIndexer {
 
   // The AlmostExact graph is used since size-1 splits and merges
   // should not affect actual index exprs.
-  const ValGraph& traversalGraph() const {
+  // Returns non-const reference because indexing may create new domains and
+  // need to update the graph.
+  ValGraph& traversalGraph() const {
```

Comment:
My horrible C++ understanding is breaking here. It returns a non-const reference but the function itself is const. How is it possible???

Reply:
Thank you. Makes sense.